packages/network-controller/package.json

```diff
@@ -33,7 +33,7 @@
     "@metamask/controller-utils": "^5.0.2",
     "@metamask/eth-json-rpc-infura": "^9.0.0",
     "@metamask/eth-json-rpc-middleware": "^12.0.0",
-    "@metamask/eth-json-rpc-provider": "^2.1.0",
+    "@metamask/eth-json-rpc-provider": "^2.2.0",
```

Comment:
Hmm. Given that we are affecting other packages in this PR, should we bump everything to ^2.2.0 first? I think it'd be better to keep this PR just focused to `eth-json-rpc-provider` if we can.

Reply:
This is necessary because of the reference path shadowing thing we discussed. Once `eth-json-rpc-provider` is moved into `packages/`, downstream packages can no longer use previous versions by pointing to `node_modules/`, because all imports from `@metamask/eth-json-rpc-provider` will always point to the path in `packages/`, and therefore to the latest version. This seems to happen even if the migrated package isn't added to the reference paths in the downstream pkgs' typescript config files.

Reply:
That makes sense, but isn't this happening because `network-controller` is currently using version 2.1.0? If we bumped that first then the latest version of the package being used and current version of the package as it exists after being migrated would be the same.

Reply:
We could bump to eth-json-rpc-provider to 2.2.0 everywhere first in a separate PR, but that PR wouldn't include the changes being made to `network-controller/` here (the last two commits). 

Because fixing the `../dist/..` import statements requires fixing that export statement in `eth-json-rpc-provider`, we'd still end up with changes being made to both packages simultaneously before the migration can be merged.

Reply:
I guess we could change the `SafeEventEmitterProvider` export statement in the original repo and release 2.2.1, but that would mean redoing the migration process from the start.